src/Generator/Passes/ResolveIncompleteDeclsPass.cs

```diff
@@ -68,7 +68,16 @@ private void EnsureCompleteDeclaration(Declaration declaration)
                 return;
 
             var @class = declaration as Class;
-            if (@class != null && @class.IsOpaque) return;
+            if (@class != null && @class.IsOpaque)
+                if (@class.Redeclarations.Count == 0)
+                    return;
+                else if (@class.Redeclarations[0] == @class && !@class.Redeclarations.Exists((Declaration c) => c.CompleteDeclaration != null))
+                {
+                    Diagnostics.Message($"There are redeclarations of class {@class.Name} at line {@class.LineNumberStart}");
```

Comment:
Remove this debugging code. This will also allow you to merge all conditions into one.